mitodl/mit-learn
# Rename channels app to learning_channels so it doesn't conflict with django channels

### What are the relevant tickets?
Indirectly related to https://github.com/mitodl/hq/issues/6126

### Description (What does it do?)
- Renames the `channels` app to  `learning_channels` so it doesn't conflict with [django channels](https://test-channels.readthedocs.io/en/latest/) which will probably be useful for streaming AI chatbot responses in a non-blocking manner (among other things).
- Adds a new dependency, `django-rename-app` which provides a convenient management command to make this transition.  The command has been added to various scripts because it needs to run before migrations are applied:
  -  heroku-release-phase.sh
  - run-django-dev.sh
  

### How can this be tested?
- Ensure you already have some existing channels created via the main branch.
- Switch to this branch and run `docker compose up`
- There should be no errors, and channel pages like `http://open.odl.local:8062/c/unit/ocw` should work as before.
- You can check the database tables to make sure that the channel tables are named `learning_channels_channel` etc instead of `channels_channel` etc
- To revert the changes, switch back to the main branch and run `./manage.py rename_app learning_channels channels`.  After that everything should still work again.
